src/endpoint/s3/ops/s3_put_bucket_policy.js

```diff
@@ -10,20 +9,11 @@ const S3Error = require('../s3_errors').S3Error;
 async function put_bucket_policy(req) {
     let policy;
     try {
-        policy = JSON.parse(JSON.stringify(req.body).toLowerCase());
+        policy = JSON.parse(JSON.stringify(req.body));
     } catch (error) {
         console.error('put_bucket_policy: Invalid JSON provided', error);
         throw new S3Error(S3Error.InvalidArgument);
     }
-    // adapting bucket policy to our schema
-    for (const statement of policy.statement) {
-        if (statement.principal) {
-            if (statement.principal.aws) statement.principal = statement.principal.aws;
-            statement.principal = _.flatten([statement.principal]);
-        }
-        if (statement.action) statement.action = _.flatten([statement.action]);
-        if (statement.resource) statement.resource = _.flatten([statement.resource]);
-    }
     await req.object_sdk.put_bucket_policy({ name: req.params.bucket, policy });
```

Comment:
Since you pass the user's policy as is in the RPC, you should catch INVALID_SCHEMA errors and return it as `MalformedPolicy`. Other wise, if the user sends a malformed policy, they will get an internal error